include/deal.II/sundials/kinsol.h

```diff
@@ -0,0 +1,664 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2017 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal.II distribution.
+//
+//---------------------------------------------------------------
+
+#ifndef dealii_sundials_kinsol_h
+#define dealii_sundials_kinsol_h
+
+#include <deal.II/base/config.h>
+#ifdef DEAL_II_WITH_SUNDIALS
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_view.h>
+#include <deal.II/lac/vector_memory.h>
+
+#include <kinsol/kinsol.h>
+#include <kinsol/kinsol_impl.h>
+#include <nvector/nvector_serial.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include <boost/signals2.hpp>
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+// Shorthand notation for KINSOL error codes.
+#define AssertKINSOL(code) Assert(code >= 0, ExcKINSOLError(code))
+
+namespace SUNDIALS
+{
+  /**
+   * Interface to SUNDIALS non linear solver (KINSOL).
+   *
+   * KINSOL is a solver for nonlinear algebraic systems. It includes a
```

Comment:
Since you use F(u) below, maybe this is the place to say "...algebraic systems $F(u)=0$."